### PR TITLE
(fix): do not expose haproxy password to UI

### PR DESCRIPTION
**Pull request by oleksandr-nc**
1. Since we need a password of at least 12 characters, we use `dummySecret123` instead of `dummySecret`
2. `checkDaemonConnection` expects to get the full Deploy Daemon configuration from the UI, so we get the password from the database if we get `dummySecret123` from the UI and the haproxy password is present in the database.
3. I added comments everywhere in the code

### CODE REVIEW

File: lib/Controller/DaemonConfigController.php [hunk 1]
```diff
@@ -59,11 +59,37 @@ public function registerDaemonConfig(array $daemonConfigParams, bool $defaultDae
 	#[PasswordConfirmationRequired]
 	public function updateDaemonConfig(string $name, array $daemonConfigParams): Response {
 		$daemonConfig = $this->daemonConfigService->getDaemonConfigByName($name);
+
+		// Safely check if "haproxy_password" exists before accessing it
+		$haproxyPassword = $daemonConfigParams['deployConfig']['haproxy_password'] ?? null;
+
+		// Restore the original password if "dummySecret123" is provided
+		if ($haproxyPassword === 'dummySecret123') {
+			$daemonConfigParams['deployConfig']['haproxy_password'] = $daemonConfig->getDeployConfig()['haproxy_password'] ?? "";
```
Comment (kyteinsky) [inline lib/Controller/DaemonConfigController.php:68]: `deploy_config` instead of `deployConfig`

Comment (oleksandr-nc) [reply lib/Controller/DaemonConfigController.php]: Keen eye, thanks for your attention!

**Review by kyteinsky — APPROVED** [review]
:rocket:

**Pull request merged**